### PR TITLE
clang bug: Dead assignment `ns_id`

### PR DESCRIPTION
**Pull request by QnJ1c2kNCg**
Remove a dead assignment of the `ns_id` variable in the
`src/nvim/api/buffer.c` file.

Refer: https://neovim.io/doc/reports/clang/report-f279da.html#EndPath

### CODE REVIEW

File: src/nvim/api/buffer.c [hunk 1]
```diff
@@ -1420,10 +1420,10 @@ Integer nvim_buf_add_highlight(Buffer buffer,
     end_line++;
   }
 
-  ns_id = extmark_add_decoration(buf, ns_id, hlg_id,
-                                 (int)line, (colnr_T)col_start,
-                                 end_line, (colnr_T)col_end,
-                                 VIRTTEXT_EMPTY);
+  extmark_add_decoration(buf, ns_id, hlg_id,
```
Comment (h-michael) [inline src/nvim/api/buffer.c]: ref: https://github.com/QnJ1c2kNCg/neovim/commit/ca1a00edd6d6345b848a28d077d6a192528f811e#diff-9c343d9e2b2c6c1a75a00e9e95432767R1423-R1427

Shouldn't this change ns_id to src_id instead of removing ns_id?

Comment (justinmk) [reply src/nvim/api/buffer.c]: good question. We have test coverage for this, but maybe there's an edge case... @bfredl should `nvim_buf_add_highlight` return the result of `extmark_add_decoration` ?

Comment (bfredl) [reply src/nvim/api/buffer.c]: `src_id` has already been calculated. This PR is correct, just delete the assigment here.
